pydantic/generics.py

```diff
@@ -78,7 +93,30 @@ def resolve_type_hint(type_: Any, typevars_map: Dict[Any, Any]) -> Type[Any]:
 
 def check_parameters_count(cls: Type[GenericModel], parameters: Tuple[Any, ...]) -> None:
     actual = len(parameters)
-    expected = len(cls.__parameters__)  # type: ignore
+    expected = len(cls.__parameters__)
     if actual != expected:
         description = 'many' if actual > expected else 'few'
         raise TypeError(f'Too {description} parameters for {cls.__name__}; actual {actual}, expected {expected}')
+
+
+def _build_generic_fields(
+    raw_fields: Dict[str, ModelField],
+    concrete_type_hints: Dict[str, Type[Any]],
+    typevars_map: Dict[TypeVarType, Type[Any]],
+) -> Dict[str, Tuple[Type[Any], FieldInfo]]:
+    return {
+        k: (_parameterize_generic_field(v, typevars_map), raw_fields[k].field_info)
+        for k, v in concrete_type_hints.items()
+        if k in raw_fields
+    }
+
+
+def _parameterize_generic_field(field_type: Type[Any], typevars_map: Dict[TypeVarType, Type[Any]]) -> Type[Any]:
+    if lenient_issubclass(field_type, GenericModel) and not field_type.__concrete__:
+        parameters = tuple(typevars_map.get(param, param) for param in field_type.__parameters__)
+        field_type = field_type[parameters]
+    return field_type
+
+
+def _is_typevar(v: Any) -> bool:
```

Comment:
there's at least one more place above this can be used where we currently have `isinstance(param, TypeVar)`

Reply:
Thanks, fixed